commit 1f30e5bb73bfafef3c66261cc0945b721b1109a5
Author: BinhNV <binhnv@exoplatform.com>
Date:   Tue Aug 7 18:37:38 2012 +0700

    commit new code

diff --git a/.classpath b/.classpath
new file mode 100644
index 0000000..8f0cce4
--- /dev/null
+++ b/.classpath
@@ -0,0 +1,235 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+  <classpathentry kind="src" path="src/main/java" including="**/*.java"/>
+  <classpathentry kind="src" path="src/main/resources" excluding="**/*.java"/>
+  <classpathentry kind="output" path="target/classes"/>
+  <classpathentry kind="var" path="M2_REPO/javax/servlet/servlet-api/3.0-alpha-1/servlet-api-3.0-alpha-1.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/portlet/portlet-api/2.0/portlet-api-2.0.jar" sourcepath="M2_REPO/javax/portlet/portlet-api/2.0/portlet-api-2.0-sources.jar">
+    <attributes>
+      <attribute value="jar:file:/home/binhnv/java/exo-dependencies/repository/javax/portlet/portlet-api/2.0/portlet-api-2.0-javadoc.jar!/" name="javadoc_location"/>
+    </attributes>
+  </classpathentry>
+  <classpathentry kind="var" path="M2_REPO/javax/faces/jsf-api/1.2_13/jsf-api-1.2_13.jar" sourcepath="M2_REPO/javax/faces/jsf-api/1.2_13/jsf-api-1.2_13-sources.jar">
+    <attributes>
+      <attribute value="jar:file:/home/binhnv/java/exo-dependencies/repository/javax/faces/jsf-api/1.2_13/jsf-api-1.2_13-javadoc.jar!/" name="javadoc_location"/>
+    </attributes>
+  </classpathentry>
+  <classpathentry kind="var" path="M2_REPO/javax/activation/activation/1.1.1/activation-1.1.1.jar" sourcepath="M2_REPO/javax/activation/activation/1.1.1/activation-1.1.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/mail/mail/1.4.4/mail-1.4.4.jar" sourcepath="M2_REPO/javax/mail/mail/1.4.4/mail-1.4.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/transaction/jta/1.1/jta-1.1.jar" sourcepath="M2_REPO/javax/transaction/jta/1.1/jta-1.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/jcr/jcr/1.0/jcr-1.0.jar" sourcepath="M2_REPO/javax/jcr/jcr/1.0/jcr-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/xml/stream/stax-api/1.0/stax-api-1.0.jar" sourcepath="M2_REPO/javax/xml/stream/stax-api/1.0/stax-api-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/xml/bind/jaxb-api/2.1/jaxb-api-2.1.jar" sourcepath="M2_REPO/javax/xml/bind/jaxb-api/2.1/jaxb-api-2.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/ws/rs/jsr311-api/1.0/jsr311-api-1.0.jar" sourcepath="M2_REPO/javax/ws/rs/jsr311-api/1.0/jsr311-api-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/inject/javax.inject/1/javax.inject-1.jar" sourcepath="M2_REPO/javax/inject/javax.inject/1/javax.inject-1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar" sourcepath="M2_REPO/javax/annotation/jsr250-api/1.0/jsr250-api-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/servlet/jsp/jsp-api/2.1/jsp-api-2.1.jar" sourcepath="M2_REPO/javax/servlet/jsp/jsp-api/2.1/jsp-api-2.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javax/ccpp/ccpp/1.0/ccpp-1.0.jar"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+  <classpathentry kind="var" path="M2_REPO/junit/junit/3.8.1/junit-3.8.1.jar" sourcepath="M2_REPO/junit/junit/3.8.1/junit-3.8.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/kernel/exo.kernel.commons/2.3.7-GA/exo.kernel.commons-2.3.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/kernel/exo.kernel.commons/2.3.7-GA/exo.kernel.commons-2.3.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/eu/medsea/mimeutil/mime-util/2.1.3/mime-util-2.1.3.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/slf4j/slf4j-api/1.5.8/slf4j-api-1.5.8.jar" sourcepath="M2_REPO/org/slf4j/slf4j-api/1.5.8/slf4j-api-1.5.8-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xpp3/xpp3/1.1.4c/xpp3-1.1.4c.jar" sourcepath="M2_REPO/xpp3/xpp3/1.1.4c/xpp3-1.1.4c-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-logging/commons-logging/1.1.1/commons-logging-1.1.1.jar" sourcepath="M2_REPO/commons-logging/commons-logging/1.1.1/commons-logging-1.1.1-sources.jar">
+    <attributes>
+      <attribute value="jar:file:/home/binhnv/java/exo-dependencies/repository/commons-logging/commons-logging/1.1.1/commons-logging-1.1.1-javadoc.jar!/" name="javadoc_location"/>
+    </attributes>
+  </classpathentry>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/kernel/exo.kernel.container/2.3.7-GA/exo.kernel.container-2.3.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/kernel/exo.kernel.container/2.3.7-GA/exo.kernel.container-2.3.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/tool/exo.tool.framework.junit/1.2.1-GA/exo.tool.framework.junit-1.2.1-GA.jar" sourcepath="M2_REPO/org/exoplatform/tool/exo.tool.framework.junit/1.2.1-GA/exo.tool.framework.junit-1.2.1-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/picocontainer/picocontainer/1.1/picocontainer-1.1.jar"/>
+  <classpathentry kind="var" path="M2_REPO/jmock/jmock/1.0.1/jmock-1.0.1.jar" sourcepath="M2_REPO/jmock/jmock/1.0.1/jmock-1.0.1-sources.jar">
+    <attributes>
+      <attribute value="jar:file:/home/binhnv/java/exo-dependencies/repository/jmock/jmock/1.0.1/jmock-1.0.1-javadoc.jar!/" name="javadoc_location"/>
+    </attributes>
+  </classpathentry>
+  <classpathentry kind="var" path="M2_REPO/xstream/xstream/1.0.2/xstream-1.0.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-beanutils/commons-beanutils/1.8.3/commons-beanutils-1.8.3.jar" sourcepath="M2_REPO/commons-beanutils/commons-beanutils/1.8.3/commons-beanutils-1.8.3-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jibx/jibx-run/1.2.2/jibx-run-1.2.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jibx/jibx-bind/1.2.2/jibx-bind-1.2.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/bcel/bcel/5.1/bcel-5.1.jar" sourcepath="M2_REPO/bcel/bcel/5.1/bcel-5.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/regexp/regexp/1.2/regexp-1.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jibx/jibx-extras/1.2.2/jibx-extras-1.2.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.organization.api/2.4.7-GA/exo.core.component.organization.api-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.organization.api/2.4.7-GA/exo.core.component.organization.api-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/kernel/exo.kernel.component.cache/2.3.7-GA/exo.kernel.component.cache-2.3.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/kernel/exo.kernel.component.cache/2.3.7-GA/exo.kernel.component.cache-2.3.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.security.core/2.4.7-GA/exo.core.component.security.core-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.security.core/2.4.7-GA/exo.core.component.security.core-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/kernel/exo.kernel.component.common/2.3.7-GA/exo.kernel.component.common-2.3.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/kernel/exo.kernel.component.common/2.3.7-GA/exo.kernel.component.common-2.3.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/quartz/quartz/1.5.2/quartz-1.5.2.jar" sourcepath="M2_REPO/quartz/quartz/1.5.2/quartz-1.5.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jgroups/jgroups/2.6.20.Final/jgroups-2.6.20.Final.jar" sourcepath="M2_REPO/org/jgroups/jgroups/2.6.20.Final/jgroups-2.6.20.Final-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-lang/commons-lang/2.6/commons-lang-2.6.jar" sourcepath="M2_REPO/commons-lang/commons-lang/2.6/commons-lang-2.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xdoclet/xdoclet-hibernate-module/1.2.3/xdoclet-hibernate-module-1.2.3.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xdoclet/xdoclet/1.2.3/xdoclet-1.2.3.jar" sourcepath="M2_REPO/xdoclet/xdoclet/1.2.3/xdoclet-1.2.3-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar" sourcepath="M2_REPO/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xdoclet/xdoclet-xdoclet-module/1.2.3/xdoclet-xdoclet-module-1.2.3.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xdoclet/xjavadoc/1.2.3/xjavadoc-1.2.3.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.portal/3.2.4-PLF-SNAPSHOT/exo.portal.component.portal-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.portal/3.2.4-PLF-SNAPSHOT/exo.portal.component.portal-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.organization.jdbc/2.4.7-GA/exo.core.component.organization.jdbc-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.organization.jdbc/2.4.7-GA/exo.core.component.organization.jdbc-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/hibernate/hibernate-core/3.3.2.GA/hibernate-core-3.3.2.GA.jar" sourcepath="M2_REPO/org/hibernate/hibernate-core/3.3.2.GA/hibernate-core-3.3.2.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/antlr/antlr/2.7.6/antlr-2.7.6.jar" sourcepath="M2_REPO/antlr/antlr/2.7.6/antlr-2.7.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar" sourcepath="M2_REPO/dom4j/dom4j/1.6.1/dom4j-1.6.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xml-apis/xml-apis/1.0.b2/xml-apis-1.0.b2.jar" sourcepath="M2_REPO/xml-apis/xml-apis/1.0.b2/xml-apis-1.0.b2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.database/2.4.7-GA/exo.core.component.database-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.database/2.4.7-GA/exo.core.component.database-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/experlog/xapool/1.5.0/xapool-1.5.0.jar" sourcepath="M2_REPO/com/experlog/xapool/1.5.0/xapool-1.5.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/hibernate/hibernate-annotations/3.4.0.GA/hibernate-annotations-3.4.0.GA.jar" sourcepath="M2_REPO/org/hibernate/hibernate-annotations/3.4.0.GA/hibernate-annotations-3.4.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/hibernate/ejb3-persistence/1.0.2.GA/ejb3-persistence-1.0.2.GA.jar" sourcepath="M2_REPO/org/hibernate/ejb3-persistence/1.0.2.GA/ejb3-persistence-1.0.2.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/hibernate/hibernate-commons-annotations/3.1.0.GA/hibernate-commons-annotations-3.1.0.GA.jar" sourcepath="M2_REPO/org/hibernate/hibernate-commons-annotations/3.1.0.GA/hibernate-commons-annotations-3.1.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-dbcp/commons-dbcp/1.2.2/commons-dbcp-1.2.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-pool/commons-pool/1.5.5/commons-pool-1.5.5.jar" sourcepath="M2_REPO/commons-pool/commons-pool/1.5.5/commons-pool-1.5.5-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/jcr/exo.jcr.component.ext/1.14.7-GA/exo.jcr.component.ext-1.14.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/jcr/exo.jcr.component.ext/1.14.7-GA/exo.jcr.component.ext-1.14.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/kernel/exo.kernel.component.command/2.3.7-GA/exo.kernel.component.command-2.3.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/kernel/exo.kernel.component.command/2.3.7-GA/exo.kernel.component.command-2.3.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-chain/commons-chain/1.2/commons-chain-1.2.jar" sourcepath="M2_REPO/commons-chain/commons-chain/1.2/commons-chain-1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-digester/commons-digester/1.8.1/commons-digester-1.8.1.jar" sourcepath="M2_REPO/commons-digester/commons-digester/1.8.1/commons-digester-1.8.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.document/2.4.7-GA/exo.core.component.document-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.document/2.4.7-GA/exo.core.component.document-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/pdfbox/pdfbox/1.6.0/pdfbox-1.6.0.jar" sourcepath="M2_REPO/org/apache/pdfbox/pdfbox/1.6.0/pdfbox-1.6.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/pdfbox/fontbox/1.6.0/fontbox-1.6.0.jar" sourcepath="M2_REPO/org/apache/pdfbox/fontbox/1.6.0/fontbox-1.6.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/pdfbox/jempbox/1.6.0/jempbox-1.6.0.jar" sourcepath="M2_REPO/org/apache/pdfbox/jempbox/1.6.0/jempbox-1.6.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/htmlparser/htmlparser/1.6/htmlparser-1.6.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/poi/poi/3.6/poi-3.6.jar" sourcepath="M2_REPO/org/apache/poi/poi/3.6/poi-3.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/poi/poi-scratchpad/3.6/poi-scratchpad-3.6.jar" sourcepath="M2_REPO/org/apache/poi/poi-scratchpad/3.6/poi-scratchpad-3.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/poi/poi-ooxml/3.6/poi-ooxml-3.6.jar" sourcepath="M2_REPO/org/apache/poi/poi-ooxml/3.6/poi-ooxml-3.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/poi/poi-ooxml-schemas/3.6/poi-ooxml-schemas-3.6.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/xmlbeans/xmlbeans/2.3.0/xmlbeans-2.3.0.jar"/>
+  <classpathentry kind="var" path="M2_REPO/stax/stax-api/1.0.1/stax-api-1.0.1.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/geronimo/specs/geronimo-stax-api_1.0_spec/1.0.1/geronimo-stax-api_1.0_spec-1.0.1.jar" sourcepath="M2_REPO/org/apache/geronimo/specs/geronimo-stax-api_1.0_spec/1.0.1/geronimo-stax-api_1.0_spec-1.0.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/tika/tika-core/0.7/tika-core-0.7.jar" sourcepath="M2_REPO/org/apache/tika/tika-core/0.7/tika-core-0.7-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/tika/tika-parsers/0.7/tika-parsers-0.7.jar" sourcepath="M2_REPO/org/apache/tika/tika-parsers/0.7/tika-parsers-0.7-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/commons/commons-compress/1.0/commons-compress-1.0.jar" sourcepath="M2_REPO/org/apache/commons/commons-compress/1.0/commons-compress-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/bouncycastle/bcmail-jdk15/1.45/bcmail-jdk15-1.45.jar" sourcepath="M2_REPO/org/bouncycastle/bcmail-jdk15/1.45/bcmail-jdk15-1.45-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/bouncycastle/bcprov-jdk15/1.45/bcprov-jdk15-1.45.jar" sourcepath="M2_REPO/org/bouncycastle/bcprov-jdk15/1.45/bcprov-jdk15-1.45-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/ccil/cowan/tagsoup/tagsoup/1.2/tagsoup-1.2.jar"/>
+  <classpathentry kind="var" path="M2_REPO/asm/asm/3.1/asm-3.1.jar" sourcepath="M2_REPO/asm/asm/3.1/asm-3.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/drewnoakes/metadata-extractor/2.4.0-beta-1/metadata-extractor-2.4.0-beta-1.jar" sourcepath="M2_REPO/com/drewnoakes/metadata-extractor/2.4.0-beta-1/metadata-extractor-2.4.0-beta-1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/jcr/exo.jcr.component.core/1.14.7-GA/exo.jcr.component.core-1.14.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/jcr/exo.jcr.component.core/1.14.7-GA/exo.jcr.component.core-1.14.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/lucene/lucene-core/2.9.4/lucene-core-2.9.4.jar" sourcepath="M2_REPO/org/apache/lucene/lucene-core/2.9.4/lucene-core-2.9.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/lucene/lucene-spellchecker/2.9.4/lucene-spellchecker-2.9.4.jar" sourcepath="M2_REPO/org/apache/lucene/lucene-spellchecker/2.9.4/lucene-spellchecker-2.9.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/lucene/lucene-memory/2.9.4/lucene-memory-2.9.4.jar" sourcepath="M2_REPO/org/apache/lucene/lucene-memory/2.9.4/lucene-memory-2.9.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/sun/xml/stream/sjsxp/1.0.1/sjsxp-1.0.1.jar"/>
+  <classpathentry kind="var" path="M2_REPO/concurrent/concurrent/1.3.4/concurrent-1.3.4.jar" sourcepath="M2_REPO/concurrent/concurrent/1.3.4/concurrent-1.3.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/jboss/jbossts/jbossjts/4.6.1.GA/jbossjts-4.6.1.GA.jar"/>
+  <classpathentry kind="var" path="M2_REPO/jboss/jbossts/jbossts-common/4.6.1.GA/jbossts-common-4.6.1.GA.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/ws/commons/ws-commons-util/1.0.1/ws-commons-util-1.0.1.jar" sourcepath="M2_REPO/org/apache/ws/commons/ws-commons-util/1.0.1/ws-commons-util-1.0.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jboss/cache/jbosscache-core/3.2.7.GA/jbosscache-core-3.2.7.GA.jar" sourcepath="M2_REPO/org/jboss/cache/jbosscache-core/3.2.7.GA/jbosscache-core-3.2.7.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jboss/javaee/jboss-transaction-api/1.0.1.GA/jboss-transaction-api-1.0.1.GA.jar" sourcepath="M2_REPO/org/jboss/javaee/jboss-transaction-api/1.0.1.GA/jboss-transaction-api-1.0.1.GA-sources.jar">
+    <attributes>
+      <attribute value="jar:file:/home/binhnv/java/exo-dependencies/repository/org/jboss/javaee/jboss-transaction-api/1.0.1.GA/jboss-transaction-api-1.0.1.GA-javadoc.jar!/" name="javadoc_location"/>
+    </attributes>
+  </classpathentry>
+  <classpathentry kind="var" path="M2_REPO/org/jboss/jboss-common-core/2.2.14.GA/jboss-common-core-2.2.14.GA.jar" sourcepath="M2_REPO/org/jboss/jboss-common-core/2.2.14.GA/jboss-common-core-2.2.14.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/jboss/logging/jboss-logging-spi/2.0.5.GA/jboss-logging-spi-2.0.5.GA.jar" sourcepath="M2_REPO/org/jboss/logging/jboss-logging-spi/2.0.5.GA/jboss-logging-spi-2.0.5.GA-sources.jar">
+    <attributes>
+      <attribute value="jar:file:/home/binhnv/java/exo-dependencies/repository/org/jboss/logging/jboss-logging-spi/2.0.5.GA/jboss-logging-spi-2.0.5.GA-javadoc.jar!/" name="javadoc_location"/>
+    </attributes>
+  </classpathentry>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.script.groovy/2.4.7-GA/exo.core.component.script.groovy-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.script.groovy/2.4.7-GA/exo.core.component.script.groovy-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/codehaus/groovy/groovy-all/1.7.6/groovy-all-1.7.6.jar" sourcepath="M2_REPO/org/codehaus/groovy/groovy-all/1.7.6/groovy-all-1.7.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/ws/exo.ws.rest.core/2.2.7-GA/exo.ws.rest.core-2.2.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/ws/exo.ws.rest.core/2.2.7-GA/exo.ws.rest.core-2.2.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/ws/exo.ws.frameworks.json/2.2.7-GA/exo.ws.frameworks.json-2.2.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/ws/exo.ws.frameworks.json/2.2.7-GA/exo.ws.frameworks.json-2.2.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/sf/jtidy/jtidy/r938/jtidy-r938.jar" sourcepath="M2_REPO/net/sf/jtidy/jtidy/r938/jtidy-r938-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-fileupload/commons-fileupload/1.2.1/commons-fileupload-1.2.1.jar" sourcepath="M2_REPO/commons-fileupload/commons-fileupload/1.2.1/commons-fileupload-1.2.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/ws/exo.ws.rest.ext/2.2.7-GA/exo.ws.rest.ext-2.2.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/ws/exo.ws.rest.ext/2.2.7-GA/exo.ws.rest.ext-2.2.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/core/exo.core.component.xml-processing/2.4.7-GA/exo.core.component.xml-processing-2.4.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/core/exo.core.component.xml-processing/2.4.7-GA/exo.core.component.xml-processing-2.4.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/ws/exo.ws.commons/2.2.7-GA/exo.ws.commons-2.2.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/ws/exo.ws.commons/2.2.7-GA/exo.ws.commons-2.2.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.common/3.2.4-PLF-SNAPSHOT/exo.portal.component.common-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.common/3.2.4-PLF-SNAPSHOT/exo.portal.component.common-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/common/common-logging/2.0.4-GA/common-logging-2.0.4-GA.jar" sourcepath="M2_REPO/org/gatein/common/common-logging/2.0.4-GA/common-logging-2.0.4-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/common/common-common/2.0.4-GA/common-common-2.0.4-GA.jar" sourcepath="M2_REPO/org/gatein/common/common-common/2.0.4-GA/common-common-2.0.4-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/kernel/exo.kernel.component.ext.cache.impl.jboss.v3/2.3.7-GA/exo.kernel.component.ext.cache.impl.jboss.v3-2.3.7-GA.jar" sourcepath="M2_REPO/org/exoplatform/kernel/exo.kernel.component.ext.cache.impl.jboss.v3/2.3.7-GA/exo.kernel.component.ext.cache.impl.jboss.v3-2.3.7-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.api/1.1.2/chromattic.api-1.1.2.jar" sourcepath="M2_REPO/org/chromattic/chromattic.api/1.1.2/chromattic.api-1.1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.spi/1.1.2/chromattic.spi-1.1.2.jar" sourcepath="M2_REPO/org/chromattic/chromattic.spi/1.1.2/chromattic.spi-1.1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.pc/3.2.4-PLF-SNAPSHOT/exo.portal.component.pc-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.pc/3.2.4-PLF-SNAPSHOT/exo.portal.component.pc-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.resources/3.2.4-PLF-SNAPSHOT/exo.portal.component.resources-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.resources/3.2.4-PLF-SNAPSHOT/exo.portal.component.resources-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/pc/pc-api/2.3.0-GA/pc-api-2.3.0-GA.jar" sourcepath="M2_REPO/org/gatein/pc/pc-api/2.3.0-GA/pc-api-2.3.0-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/wci/wci-wci/2.1.0-GA/wci-wci-2.1.0-GA.jar" sourcepath="M2_REPO/org/gatein/wci/wci-wci/2.1.0-GA/wci-wci-2.1.0-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/pc/pc-portlet/2.3.0-GA/pc-portlet-2.3.0-GA.jar" sourcepath="M2_REPO/org/gatein/pc/pc-portlet/2.3.0-GA/pc-portlet-2.3.0-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/staxnav/staxnav.core/0.9.6/staxnav.core-0.9.6.jar" sourcepath="M2_REPO/org/staxnav/staxnav.core/0.9.6/staxnav.core-0.9.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/pc/pc-federation/2.3.0-GA/pc-federation-2.3.0-GA.jar" sourcepath="M2_REPO/org/gatein/pc/pc-federation/2.3.0-GA/pc-federation-2.3.0-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/pc/pc-bridge/2.3.0-GA/pc-bridge-2.3.0-GA.jar" sourcepath="M2_REPO/org/gatein/pc/pc-bridge/2.3.0-GA/pc-bridge-2.3.0-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/portals/bridges/portals-bridges-common/1.0.4/portals-bridges-common-1.0.4.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.identity/3.2.4-PLF-SNAPSHOT/exo.portal.component.identity-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.identity/3.2.4-PLF-SNAPSHOT/exo.portal.component.identity-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-core/1.3.0.GA/picketlink-idm-core-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-core/1.3.0.GA/picketlink-idm-core-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-common/1.3.0.GA/picketlink-idm-common-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-common/1.3.0.GA/picketlink-idm-common-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/log4j/log4j/1.2.14/log4j-1.2.14.jar" sourcepath="M2_REPO/log4j/log4j/1.2.14/log4j-1.2.14-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-api/1.3.0.GA/picketlink-idm-api-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-api/1.3.0.GA/picketlink-idm-api-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-spi/1.3.0.GA/picketlink-idm-spi-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-spi/1.3.0.GA/picketlink-idm-spi-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/sun/xml/bind/jaxb-impl/2.1.8/jaxb-impl-2.1.8.jar" sourcepath="M2_REPO/com/sun/xml/bind/jaxb-impl/2.1.8/jaxb-impl-2.1.8-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-hibernate/1.3.0.GA/picketlink-idm-hibernate-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-hibernate/1.3.0.GA/picketlink-idm-hibernate-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/javassist/javassist/3.8.0.GA/javassist-3.8.0.GA.jar" sourcepath="M2_REPO/javassist/javassist/3.8.0.GA/javassist-3.8.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/hibernate/hibernate-cglib-repack/2.1_3/hibernate-cglib-repack-2.1_3.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-ldap/1.3.0.GA/picketlink-idm-ldap-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-ldap/1.3.0.GA/picketlink-idm-ldap-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/picketlink/idm/picketlink-idm-cache/1.3.0.GA/picketlink-idm-cache-1.3.0.GA.jar" sourcepath="M2_REPO/org/picketlink/idm/picketlink-idm-cache/1.3.0.GA/picketlink-idm-cache-1.3.0.GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/mop/mop-api/1.1.1-GA/mop-api-1.1.1-GA.jar" sourcepath="M2_REPO/org/gatein/mop/mop-api/1.1.1-GA/mop-api-1.1.1-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/mop/mop-spi/1.1.1-GA/mop-spi-1.1.1-GA.jar" sourcepath="M2_REPO/org/gatein/mop/mop-spi/1.1.1-GA/mop-spi-1.1.1-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/mop/mop-core/1.1.1-GA/mop-core-1.1.1-GA.jar" sourcepath="M2_REPO/org/gatein/mop/mop-core/1.1.1-GA/mop-core-1.1.1-GA-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.ext/1.1.2/chromattic.ext-1.1.2.jar" sourcepath="M2_REPO/org/chromattic/chromattic.ext/1.1.2/chromattic.ext-1.1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.common/1.1.2/chromattic.common-1.1.2.jar" sourcepath="M2_REPO/org/chromattic/chromattic.common/1.1.2/chromattic.common-1.1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.apt/1.1.2/chromattic.apt-1.1.2.jar" sourcepath="M2_REPO/org/chromattic/chromattic.apt/1.1.2/chromattic.apt-1.1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.metamodel/1.1.2/chromattic.metamodel-1.1.2.jar" sourcepath="M2_REPO/org/chromattic/chromattic.metamodel/1.1.2/chromattic.metamodel-1.1.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.testgenerator/1.1.1/chromattic.testgenerator-1.1.1.jar" sourcepath="M2_REPO/org/chromattic/chromattic.testgenerator/1.1.1/chromattic.testgenerator-1.1.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/google/code/javaparser/javaparser/1.0.8/javaparser-1.0.8.jar" sourcepath="M2_REPO/com/google/code/javaparser/javaparser/1.0.8/javaparser-1.0.8-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/reflext/reflext.api/1.1.0/reflext.api-1.1.0.jar" sourcepath="M2_REPO/org/reflext/reflext.api/1.1.0/reflext.api-1.1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/reflext/reflext.spi/1.1.0/reflext.spi-1.1.0.jar" sourcepath="M2_REPO/org/reflext/reflext.spi/1.1.0/reflext.spi-1.1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/reflext/reflext.core/1.1.0/reflext.core-1.1.0.jar" sourcepath="M2_REPO/org/reflext/reflext.core/1.1.0/reflext.core-1.1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/reflext/reflext.jlr/1.1.0/reflext.jlr-1.1.0.jar" sourcepath="M2_REPO/org/reflext/reflext.jlr/1.1.0/reflext.jlr-1.1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/reflext/reflext.apt/1.1.0/reflext.apt-1.1.0.jar" sourcepath="M2_REPO/org/reflext/reflext.apt/1.1.0/reflext.apt-1.1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/json/json/20070829/json-20070829.jar" sourcepath="M2_REPO/org/json/json/20070829/json-20070829-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.web.controller/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.controller-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.web.controller/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.controller-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.web.resources/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.resources-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.web.resources/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.resources-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-io/commons-io/1.4/commons-io-1.4.jar" sourcepath="M2_REPO/commons-io/commons-io/1.4/commons-io-1.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.webui.framework/3.2.4-PLF-SNAPSHOT/exo.portal.webui.framework-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.webui.framework/3.2.4-PLF-SNAPSHOT/exo.portal.webui.framework-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.scripting/3.2.4-PLF-SNAPSHOT/exo.portal.component.scripting-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.scripting/3.2.4-PLF-SNAPSHOT/exo.portal.component.scripting-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.webui.core/3.2.4-PLF-SNAPSHOT/exo.portal.webui.core-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.webui.core/3.2.4-PLF-SNAPSHOT/exo.portal.webui.core-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.webui.portlet/3.2.4-PLF-SNAPSHOT/exo.portal.webui.portlet-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.webui.portlet/3.2.4-PLF-SNAPSHOT/exo.portal.webui.portlet-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.web.server/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.server-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.web.server/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.server-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/captcha/simplecaptcha/1.1.1-GA-Patch01/simplecaptcha-1.1.1-GA-Patch01.jar" sourcepath="M2_REPO/org/gatein/captcha/simplecaptcha/1.1.1-GA-Patch01/simplecaptcha-1.1.1-GA-Patch01-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/jhlabs/filters/2.0.235/filters-2.0.235.jar" sourcepath="M2_REPO/com/jhlabs/filters/2.0.235/filters-2.0.235-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.web.api/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.api-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.web.api/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.api-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.webui.eXo/3.2.4-PLF-SNAPSHOT/exo.portal.webui.eXo-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.webui.eXo/3.2.4-PLF-SNAPSHOT/exo.portal.webui.eXo-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.application-registry/3.2.4-PLF-SNAPSHOT/exo.portal.component.application-registry-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.application-registry/3.2.4-PLF-SNAPSHOT/exo.portal.component.application-registry-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/shindig/shindig-gadgets/2.0.2/shindig-gadgets-2.0.2.jar" sourcepath="M2_REPO/org/gatein/shindig/shindig-gadgets/2.0.2/shindig-gadgets-2.0.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/shindig/shindig-common/2.0.2/shindig-common-2.0.2.jar" sourcepath="M2_REPO/org/gatein/shindig/shindig-common/2.0.2/shindig-common-2.0.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/google/inject/guice/2.0/guice-2.0.jar"/>
+  <classpathentry kind="var" path="M2_REPO/aopalliance/aopalliance/1.0/aopalliance-1.0.jar" sourcepath="M2_REPO/aopalliance/aopalliance/1.0/aopalliance-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/google/inject/extensions/guice-jmx/2.0/guice-jmx-2.0.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/google/guava/guava/r09/guava-r09.jar" sourcepath="M2_REPO/com/google/guava/guava/r09/guava-r09-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-codec/commons-codec/1.4/commons-codec-1.4.jar" sourcepath="M2_REPO/commons-codec/commons-codec/1.4/commons-codec-1.4-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/joda-time/joda-time/1.6/joda-time-1.6.jar" sourcepath="M2_REPO/joda-time/joda-time/1.6/joda-time-1.6-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/oauth/core/oauth/20100527/oauth-20100527.jar" sourcepath="M2_REPO/net/oauth/core/oauth/20100527/oauth-20100527-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-betwixt/commons-betwixt/0.8/commons-betwixt-0.8.jar"/>
+  <classpathentry kind="var" path="M2_REPO/commons-beanutils/commons-beanutils-core/1.7.0/commons-beanutils-core-1.7.0.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/thoughtworks/xstream/xstream/1.3.1/xstream-1.3.1.jar" sourcepath="M2_REPO/com/thoughtworks/xstream/xstream/1.3.1/xstream-1.3.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xpp3/xpp3_min/1.1.4c/xpp3_min-1.1.4c.jar" sourcepath="M2_REPO/xpp3/xpp3_min/1.1.4c/xpp3_min-1.1.4c-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/sf/ehcache/ehcache-core/2.2.0/ehcache-core-2.2.0.jar" sourcepath="M2_REPO/net/sf/ehcache/ehcache-core/2.2.0/ehcache-core-2.2.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/de/odysseus/juel/juel-impl/2.2.3/juel-impl-2.2.3.jar" sourcepath="M2_REPO/de/odysseus/juel/juel-impl/2.2.3/juel-impl-2.2.3-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/tomcat/el-api/6.0.29/el-api-6.0.29.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/tomcat/jasper-el/6.0.29/jasper-el-6.0.29.jar"/>
+  <classpathentry kind="var" path="M2_REPO/caja/caja/r4251/caja-r4251.jar"/>
+  <classpathentry kind="var" path="M2_REPO/caja/htmlparser/r4209/htmlparser-r4209.jar"/>
+  <classpathentry kind="var" path="M2_REPO/xerces/xercesImpl/2.9.1/xercesImpl-2.9.1.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/googlecode/json-simple/json-simple/1.1/json-simple-1.1.jar" sourcepath="M2_REPO/com/googlecode/json-simple/json-simple/1.1/json-simple-1.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/oauth/core/oauth-httpclient4/20090913/oauth-httpclient4-20090913.jar" sourcepath="M2_REPO/net/oauth/core/oauth-httpclient4/20090913/oauth-httpclient4-20090913-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/oauth/core/oauth-consumer/20090617/oauth-consumer-20090617.jar" sourcepath="M2_REPO/net/oauth/core/oauth-consumer/20090617/oauth-consumer-20090617-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/httpcomponents/httpclient/4.0.1/httpclient-4.0.1.jar" sourcepath="M2_REPO/org/apache/httpcomponents/httpclient/4.0.1/httpclient-4.0.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/httpcomponents/httpcore/4.0.1/httpcore-4.0.1.jar" sourcepath="M2_REPO/org/apache/httpcomponents/httpcore/4.0.1/httpcore-4.0.1-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/oauth/core/oauth-provider/20100527/oauth-provider-20100527.jar" sourcepath="M2_REPO/net/oauth/core/oauth-provider/20100527/oauth-provider-20100527-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/google/inject/extensions/guice-multibindings/2.0/guice-multibindings-2.0.jar"/>
+  <classpathentry kind="var" path="M2_REPO/rome/rome/1.0/rome-1.0.jar" sourcepath="M2_REPO/rome/rome/1.0/rome-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/jdom/jdom/1.0/jdom-1.0.jar" sourcepath="M2_REPO/jdom/jdom/1.0/jdom-1.0-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/rome/modules/0.3.2/modules-0.3.2.jar" sourcepath="M2_REPO/rome/modules/0.3.2/modules-0.3.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/com/ibm/icu/icu4j/3.8/icu4j-3.8.jar"/>
+  <classpathentry kind="var" path="M2_REPO/net/sourceforge/nekohtml/nekohtml/1.9.14/nekohtml-1.9.14.jar" sourcepath="M2_REPO/net/sourceforge/nekohtml/nekohtml/1.9.14/nekohtml-1.9.14-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/apache/sanselan/sanselan/0.97-incubator/sanselan-0.97-incubator.jar" sourcepath="M2_REPO/org/apache/sanselan/sanselan/0.97-incubator/sanselan-0.97-incubator-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.webui.dashboard/3.2.4-PLF-SNAPSHOT/exo.portal.webui.dashboard-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.webui.dashboard/3.2.4-PLF-SNAPSHOT/exo.portal.webui.dashboard-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.webui.portal/3.2.4-PLF-SNAPSHOT/exo.portal.webui.portal-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.webui.portal/3.2.4-PLF-SNAPSHOT/exo.portal.webui.portal-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.gadgets-core/3.2.4-PLF-SNAPSHOT/exo.portal.gadgets-core-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.gadgets-core/3.2.4-PLF-SNAPSHOT/exo.portal.gadgets-core-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/portal/exo.portal.component.web.security/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.security-3.2.4-PLF-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/portal/exo.portal.component.web.security/3.2.4-PLF-SNAPSHOT/exo.portal.component.web.security-3.2.4-PLF-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/shindig/shindig-features/2.0.2/shindig-features-2.0.2.jar" sourcepath="M2_REPO/org/gatein/shindig/shindig-features/2.0.2/shindig-features-2.0.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/gatein/shindig/shindig-social-api/2.0.2/shindig-social-api-2.0.2.jar" sourcepath="M2_REPO/org/gatein/shindig/shindig-social-api/2.0.2/shindig-social-api-2.0.2-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/exoplatform/jcr-exo/1.0-SNAPSHOT/jcr-exo-1.0-SNAPSHOT.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/hsqldb/hsqldb/2.0.0/hsqldb-2.0.0.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/social/exo.social.component.core/1.2.9-SNAPSHOT/exo.social.component.core-1.2.9-SNAPSHOT.jar" sourcepath="M2_REPO/org/exoplatform/social/exo.social.component.core/1.2.9-SNAPSHOT/exo.social.component.core-1.2.9-SNAPSHOT-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/exoplatform/doc/doc-style/1.0.6-SNAPSHOT/doc-style-1.0.6-SNAPSHOT-templates.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/wikbook/wikbook.template.core/0.9.37/wikbook.template.core-0.9.37.jar" sourcepath="M2_REPO/org/wikbook/wikbook.template.core/0.9.37/wikbook.template.core-0.9.37-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/freemarker/freemarker/2.3.18/freemarker-2.3.18.jar" sourcepath="M2_REPO/org/freemarker/freemarker/2.3.18/freemarker-2.3.18-sources.jar"/>
+  <classpathentry kind="var" path="M2_REPO/org/chromattic/chromattic.core/1.1.1/chromattic.core-1.1.1.jar" sourcepath="M2_REPO/org/chromattic/chromattic.core/1.1.1/chromattic.core-1.1.1-sources.jar"/>
+  <classpathentry kind="lib" path="/home/binhnv/java/jdk1.6/lib/tools.jar"/>
+</classpath>
\ No newline at end of file
diff --git a/.project b/.project
new file mode 100644
index 0000000..9449fae
--- /dev/null
+++ b/.project
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+  <name>jcr-exo-portlet</name>
+  <comment>NO_M2ECLIPSE_SUPPORT: Project files created with the maven-eclipse-plugin are not supported in M2Eclipse.</comment>
+  <projects/>
+  <buildSpec>
+    <buildCommand>
+      <name>org.eclipse.jdt.core.javabuilder</name>
+    </buildCommand>
+  </buildSpec>
+  <natures>
+    <nature>org.eclipse.jdt.core.javanature</nature>
+  </natures>
+</projectDescription>
\ No newline at end of file
diff --git a/pom.xml b/pom.xml
new file mode 100644
index 0000000..54b7584
--- /dev/null
+++ b/pom.xml
@@ -0,0 +1,142 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>exoplatform</groupId>
+  <artifactId>jcr-exo-portlet</artifactId>
+  <packaging>war</packaging>
+  <version>1.0-SNAPSHOT</version>
+  <name>jcr-exo-portlet Maven Webapp</name>
+  <url>http://maven.apache.org</url>
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>3.8.1</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+  	<groupId>javax.servlet</groupId>
+  	<artifactId>servlet-api</artifactId>
+  	<version>3.0-alpha-1</version>
+		<scope>provided</scope>
+	</dependency>
+
+  <dependency>
+  	<groupId>javax.portlet</groupId>
+  	<artifactId>portlet-api</artifactId>
+  	<version>2.0</version>
+		<scope>provided</scope>
+	</dependency>
+
+  <dependency>
+  	<groupId>org.exoplatform.kernel</groupId>
+  	<artifactId>exo.kernel.commons</artifactId>
+  	<version>2.3.7-GA</version>
+		<scope>provided</scope>
+	</dependency>
+
+  <dependency>
+ 		<groupId>org.exoplatform.kernel</groupId>
+  	<artifactId>exo.kernel.container</artifactId>
+  	<version>2.3.7-GA</version>
+		<scope>provided</scope>
+	</dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.core</groupId>
+      <artifactId>exo.core.component.organization.api</artifactId>
+			<version>2.4.7-GA</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.core</groupId>
+      <artifactId>exo.core.component.security.core</artifactId>
+		 <version>2.4.7-GA</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.component.portal</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.component.resources</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.component.web.controller</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.webui.framework</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.webui.core</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.webui.eXo</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.webui.portlet</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.webui.dashboard</artifactId>
+ 		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.exoplatform.portal</groupId>
+      <artifactId>exo.portal.webui.portal</artifactId>
+  		<version>3.2.4-PLF-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>exoplatform</groupId>
+      <artifactId>jcr-exo</artifactId>
+  		<version>1.0-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+    
+    <dependency>
+    	<groupId>org.exoplatform.social</groupId>
+  		<artifactId>exo.social.component.core</artifactId>
+  		<version>1.2.9-SNAPSHOT</version>
+    	<scope>provided</scope>
+    </dependency>
+  </dependencies>
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+  <build>
+    <finalName>jcr-exo-portlet</finalName>
+  </build>
+</project>
diff --git a/src/main/java/exoplatform/bookstore/portlet/UIBookManagementPortlet.java b/src/main/java/exoplatform/bookstore/portlet/UIBookManagementPortlet.java
new file mode 100644
index 0000000..631fae4
--- /dev/null
+++ b/src/main/java/exoplatform/bookstore/portlet/UIBookManagementPortlet.java
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2003-2012 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package exoplatform.bookstore.portlet;
+
+import org.exoplatform.webui.config.annotation.ComponentConfig;
+import org.exoplatform.webui.core.UIPortletApplication;
+import org.exoplatform.webui.core.lifecycle.UIApplicationLifecycle;
+
+import exoplatform.bookstore.webui.UIBookManagement;
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : BinhNV
+ *          binhnv@exoplatform.com
+ * Aug 6, 2012  
+ */
+@ComponentConfig(
+    lifecycle = UIApplicationLifecycle.class,
+    template = "app:/groovy/portlet/UIBookManagementPortlet.gtmpl"
+)
+public class UIBookManagementPortlet extends UIPortletApplication {
+
+  public UIBookManagementPortlet() throws Exception {
+    addChild(UIBookManagement.class, null, null);
+  }
+
+}
diff --git a/src/main/java/exoplatform/bookstore/service/BookstoreServiceUtil.java b/src/main/java/exoplatform/bookstore/service/BookstoreServiceUtil.java
new file mode 100644
index 0000000..cd42438
--- /dev/null
+++ b/src/main/java/exoplatform/bookstore/service/BookstoreServiceUtil.java
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2003-2012 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package exoplatform.bookstore.service;
+
+import java.util.List;
+
+import org.exoplatform.container.ExoContainer;
+import org.exoplatform.container.ExoContainerContext;
+
+import exoplatform.BookStoreService;
+import exoplatform.entity.Book;
+import exoplatform.exception.BookNotFoundException;
+
+
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : BinhNV
+ *          binhnv@exoplatform.com
+ * Aug 6, 2012  
+ */
+public class BookstoreServiceUtil {
+  
+  public static BookStoreService getBookstoreService() {
+    ExoContainer container = ExoContainerContext.getCurrentContainer();
+    return (BookStoreService) container.getComponentInstanceOfType(BookStoreService.class);
+  }
+  
+  public static List<Book> getAllBook() {
+    return getBookstoreService().getAllBook();
+  }
+  
+  public void deleteBook(String bookId) throws BookNotFoundException {
+    getBookstoreService().deleteBook(bookId);
+  }
+  
+}
diff --git a/src/main/java/exoplatform/bookstore/webui/UIBookList.java b/src/main/java/exoplatform/bookstore/webui/UIBookList.java
new file mode 100644
index 0000000..ae01c81
--- /dev/null
+++ b/src/main/java/exoplatform/bookstore/webui/UIBookList.java
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2003-2012 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package exoplatform.bookstore.webui;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.exoplatform.webui.application.WebuiRequestContext;
+import org.exoplatform.webui.config.annotation.ComponentConfig;
+import org.exoplatform.webui.config.annotation.EventConfig;
+import org.exoplatform.webui.core.UIComponent;
+import org.exoplatform.webui.core.UIContainer;
+import org.exoplatform.webui.core.lifecycle.UIContainerLifecycle;
+import org.exoplatform.webui.core.lifecycle.UIFormLifecycle;
+import org.exoplatform.webui.event.Event;
+import org.exoplatform.webui.event.EventListener;
+import org.exoplatform.webui.form.UIForm;
+
+import exoplatform.BookStoreService;
+import exoplatform.bookstore.service.BookstoreServiceUtil;
+import exoplatform.entity.Book;
+import exoplatform.exception.BookNotFoundException;
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : BinhNV
+ *          binhnv@exoplatform.com
+ * Aug 6, 2012  
+ */
+@ComponentConfig(
+    template = "app:/groovy/webui/UIBookList.gtmpl",
+    events = {
+      @EventConfig(listeners=UIBookList.EditActionListener.class),
+      @EventConfig(listeners=UIBookList.DeleteActionListener.class)
+    }
+)
+public class UIBookList extends UIComponent {
+
+  private List<Book> books = new ArrayList<Book>();
+  
+  public static class EditActionListener extends EventListener<UIBookList> {
+
+    @Override
+    public void execute(Event<UIBookList> event) throws Exception {
+      WebuiRequestContext ctx = event.getRequestContext();
+      String bookId = ctx.getRequestParameter(OBJECTID);
+      
+    }
+    
+  }
+  
+  public static class DeleteActionListener extends EventListener<UIBookList> {
+
+    @Override
+    public void execute(Event<UIBookList> event) throws Exception {
+      WebuiRequestContext ctx = event.getRequestContext();
+      String bookId = ctx.getRequestParameter(OBJECTID);
+      try {
+        BookStoreService service = BookstoreServiceUtil.getBookstoreService();
+        service.deleteBook(bookId);
+      } catch (BookNotFoundException e) {
+        e.printStackTrace();
+      }
+      ctx.addUIComponentToUpdateByAjax(event.getSource());
+    }
+    
+  }
+  
+  /**
+   * @return the books
+   */
+  public List<Book> getBooks() {
+    books = BookstoreServiceUtil.getAllBook(); 
+    return books;
+  }
+
+  /**
+   * @param books the books to set
+   */
+  public void setBooks(List<Book> books) {
+    this.books = books;
+  }
+  
+}
diff --git a/src/main/java/exoplatform/bookstore/webui/UIBookManagement.java b/src/main/java/exoplatform/bookstore/webui/UIBookManagement.java
new file mode 100644
index 0000000..37f2ba4
--- /dev/null
+++ b/src/main/java/exoplatform/bookstore/webui/UIBookManagement.java
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2003-2012 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package exoplatform.bookstore.webui;
+
+import org.exoplatform.webui.config.annotation.ComponentConfig;
+import org.exoplatform.webui.core.UIContainer;
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : BinhNV
+ *          binhnv@exoplatform.com
+ * Aug 6, 2012  
+ */
+@ComponentConfig(
+    template = "app:/groovy/webui/UIBookManagement.gtmpl"
+    
+)
+public class UIBookManagement extends UIContainer {
+
+  public UIBookManagement() throws Exception {
+    addChild(UIBookList.class, null, null);
+  }
+  
+}
diff --git a/src/main/webapp/WEB-INF/conf/portlet/webui/configuration.xml b/src/main/webapp/WEB-INF/conf/portlet/webui/configuration.xml
new file mode 100644
index 0000000..c1d1569
--- /dev/null
+++ b/src/main/webapp/WEB-INF/conf/portlet/webui/configuration.xml
@@ -0,0 +1,27 @@
+<!--
+
+    Copyright (C) 2003-2011 eXo Platform SAS.
+
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Affero General Public License
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see<http://www.gnu.org/licenses/>.
+
+-->
+<webui-configuration>
+  <application>
+    <ui-component-root>exoplatform.bookstore.portlet.UIBookManagementPortlet</ui-component-root>
+    <state-manager>org.exoplatform.webui.application.portlet.ParentAppStateManager</state-manager>
+    <application-lifecycle-listeners>       
+      <listener>org.exoplatform.portal.application.PortletStatisticLifecycle</listener>
+    </application-lifecycle-listeners>
+  </application>
+</webui-configuration>
\ No newline at end of file
diff --git a/src/main/webapp/WEB-INF/portlet.xml b/src/main/webapp/WEB-INF/portlet.xml
new file mode 100644
index 0000000..0110c5e
--- /dev/null
+++ b/src/main/webapp/WEB-INF/portlet.xml
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2003-2011 eXo Platform SAS.
+
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Affero General Public License
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see<http://www.gnu.org/licenses/>.
+
+-->
+<portlet-app
+  xmlns="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd" version="1.0">
+  
+	<portlet>
+	    <description xml:lang="EN">Bookstore Portlet</description>
+	    <portlet-name>BookstorePortlet</portlet-name>
+	    <display-name xml:lang="EN">Bookstore Portlet</display-name>
+	    <portlet-class>org.exoplatform.webui.application.portlet.PortletApplicationController</portlet-class>
+	    <init-param>
+	      <name>webui.configuration</name>
+	      <value>/WEB-INF/conf/portlet/webui/configuration.xml</value>
+	    </init-param>
+	    <expiration-cache>0</expiration-cache>
+	    <supports>
+	      <mime-type>text/html</mime-type>
+	    </supports>
+	    <supported-locale>en</supported-locale>
+	    <portlet-info>
+	      <title>Bookstore Portlet</title>
+	      <short-title>Bookstore Portlet</short-title>
+	      <keywords>bookstore</keywords>
+	    </portlet-info>
+	    <portlet-preferences>
+	      <preference>
+	        <name>template</name>
+	        <value>app:/groovy/portlet/UIBookManagemenPortlet.gtmpl</value>
+	        <read-only>false</read-only>
+	      </preference>
+	    </portlet-preferences>
+  </portlet>    
+    
+</portlet-app>
\ No newline at end of file
diff --git a/src/main/webapp/WEB-INF/web.xml b/src/main/webapp/WEB-INF/web.xml
new file mode 100644
index 0000000..9455772
--- /dev/null
+++ b/src/main/webapp/WEB-INF/web.xml
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+
+    Copyright (C) 2003-2011 eXo Platform SAS.
+
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Affero General Public License
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see<http://www.gnu.org/licenses/>.
+
+-->
+<!DOCTYPE web-app PUBLIC
+ "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN"
+ "http://java.sun.com/dtd/web-app_2_3.dtd" >
+
+<web-app>
+    
+  <display-name>BookstorePortlet</display-name>
+  
+  <description>
+    Bookstore
+  </description>
+  
+  <filter>
+    <filter-name>ResourceRequestFilter</filter-name>
+    <filter-class>org.exoplatform.portal.application.ResourceRequestFilter</filter-class>
+  </filter>
+
+  <filter-mapping>
+    <filter-name>ResourceRequestFilter</filter-name>
+    <url-pattern>/*</url-pattern>
+  </filter-mapping>
+
+  <servlet>
+    <servlet-name>GateInServlet</servlet-name>
+    <servlet-class>org.gatein.wci.api.GateInServlet</servlet-class>
+    <load-on-startup>0</load-on-startup>
+  </servlet>
+  
+  <servlet-mapping>
+    <servlet-name>GateInServlet</servlet-name>
+    <url-pattern>/gateinservlet</url-pattern>
+  </servlet-mapping>
+  
+</web-app>
diff --git a/src/main/webapp/groovy/portlet/UIBookManagementPortlet.gtmpl b/src/main/webapp/groovy/portlet/UIBookManagementPortlet.gtmpl
new file mode 100644
index 0000000..1d9dd63
--- /dev/null
+++ b/src/main/webapp/groovy/portlet/UIBookManagementPortlet.gtmpl
@@ -0,0 +1,3 @@
+<div id="<% uicomponent.getId(); %>">
+	<% uicomponent.renderChildren(); %>
+</div>
\ No newline at end of file
diff --git a/src/main/webapp/groovy/webui/UIBookList.gtmpl b/src/main/webapp/groovy/webui/UIBookList.gtmpl
new file mode 100644
index 0000000..c79a1a5
--- /dev/null
+++ b/src/main/webapp/groovy/webui/UIBookList.gtmpl
@@ -0,0 +1,59 @@
+<%
+	import org.exoplatform.webui.core.UIPopupWindow;
+	import org.exoplatform.portal.webui.util.Util;
+	import org.exoplatform.social.core.service.LinkProvider;
+%>
+
+<%
+	def pcontext = Util.getPortalRequestContext();
+	def jsManager = pcontext.getJavascriptManager();
+	jsManager.importJavascript("eXo.social.Util", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+	jsManager.importJavascript("eXo.social.PortalHttpRequest", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+	jsManager.importJavascript("eXo.social.webui.UIActivity", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+%>
+
+<div id="$uicomponent.id">
+
+	<%
+		def bookList = uicomponent.getBooks();
+		if (bookList.isEmpty()) {
+	%>
+		<div>No resource</div>
+	<%
+		} else {
+	%>
+			<table border="1">
+					<tr>
+						<th>Book Name</th>
+						<th>Book Category</th>
+						<th>Action</th>
+					</tr>
+				<%
+					for (book in bookList) {
+						def bookId = book.getBookId();
+						def name = book.getName();
+						def category = book.getCategory().toString();
+						def content = book.getContent();
+						def deleteBookAction = uicomponent.event("Delete", bookId) + "; return false"
+				%>
+						<tr>
+							<td>
+								$name
+							</td>
+							<td align="center">
+								$category
+							</td>
+							<td align="center">
+								<input type="button" name="btnEdit" value="Edit"/>
+								<input type="button" name="btnDelete" value="Delete" onclick="$deleteBookAction" class="Action"/>
+							</td>
+						</tr>
+				<%	
+					}
+				%>
+			</table>
+	<%
+		}
+	%>
+
+</div>
\ No newline at end of file
diff --git a/src/main/webapp/groovy/webui/UIBookManagement.gtmpl b/src/main/webapp/groovy/webui/UIBookManagement.gtmpl
new file mode 100644
index 0000000..e623816
--- /dev/null
+++ b/src/main/webapp/groovy/webui/UIBookManagement.gtmpl
@@ -0,0 +1,3 @@
+<div id="$uicomponent.id">
+<% uicomponent.renderChildren(); %>
+</div>
\ No newline at end of file
diff --git a/src/main/webapp/index.jsp b/src/main/webapp/index.jsp
new file mode 100644
index 0000000..c38169b
--- /dev/null
+++ b/src/main/webapp/index.jsp
@@ -0,0 +1,5 @@
+<html>
+<body>
+<h2>Hello World!</h2>
+</body>
+</html>
diff --git a/target/classes/exoplatform/bookstore/portlet/UIBookManagementPortlet.class b/target/classes/exoplatform/bookstore/portlet/UIBookManagementPortlet.class
new file mode 100644
index 0000000..1e645d4
Binary files /dev/null and b/target/classes/exoplatform/bookstore/portlet/UIBookManagementPortlet.class differ
diff --git a/target/classes/exoplatform/bookstore/service/BookstoreServiceUtil.class b/target/classes/exoplatform/bookstore/service/BookstoreServiceUtil.class
new file mode 100644
index 0000000..02bda27
Binary files /dev/null and b/target/classes/exoplatform/bookstore/service/BookstoreServiceUtil.class differ
diff --git a/target/classes/exoplatform/bookstore/webui/UIBookList$DeleteActionListener.class b/target/classes/exoplatform/bookstore/webui/UIBookList$DeleteActionListener.class
new file mode 100644
index 0000000..8b993e8
Binary files /dev/null and b/target/classes/exoplatform/bookstore/webui/UIBookList$DeleteActionListener.class differ
diff --git a/target/classes/exoplatform/bookstore/webui/UIBookList$EditActionListener.class b/target/classes/exoplatform/bookstore/webui/UIBookList$EditActionListener.class
new file mode 100644
index 0000000..77f0bac
Binary files /dev/null and b/target/classes/exoplatform/bookstore/webui/UIBookList$EditActionListener.class differ
diff --git a/target/classes/exoplatform/bookstore/webui/UIBookList.class b/target/classes/exoplatform/bookstore/webui/UIBookList.class
new file mode 100644
index 0000000..86ff68c
Binary files /dev/null and b/target/classes/exoplatform/bookstore/webui/UIBookList.class differ
diff --git a/target/classes/exoplatform/bookstore/webui/UIBookManagement.class b/target/classes/exoplatform/bookstore/webui/UIBookManagement.class
new file mode 100644
index 0000000..befae1f
Binary files /dev/null and b/target/classes/exoplatform/bookstore/webui/UIBookManagement.class differ
diff --git a/target/jcr-exo-portlet.war b/target/jcr-exo-portlet.war
new file mode 100644
index 0000000..baaad1b
Binary files /dev/null and b/target/jcr-exo-portlet.war differ
diff --git a/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/portlet/UIBookManagementPortlet.class b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/portlet/UIBookManagementPortlet.class
new file mode 100644
index 0000000..1e645d4
Binary files /dev/null and b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/portlet/UIBookManagementPortlet.class differ
diff --git a/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/service/BookstoreServiceUtil.class b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/service/BookstoreServiceUtil.class
new file mode 100644
index 0000000..02bda27
Binary files /dev/null and b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/service/BookstoreServiceUtil.class differ
diff --git a/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList$DeleteActionListener.class b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList$DeleteActionListener.class
new file mode 100644
index 0000000..8b993e8
Binary files /dev/null and b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList$DeleteActionListener.class differ
diff --git a/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList$EditActionListener.class b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList$EditActionListener.class
new file mode 100644
index 0000000..77f0bac
Binary files /dev/null and b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList$EditActionListener.class differ
diff --git a/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList.class b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList.class
new file mode 100644
index 0000000..86ff68c
Binary files /dev/null and b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookList.class differ
diff --git a/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookManagement.class b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookManagement.class
new file mode 100644
index 0000000..befae1f
Binary files /dev/null and b/target/jcr-exo-portlet/WEB-INF/classes/exoplatform/bookstore/webui/UIBookManagement.class differ
diff --git a/target/jcr-exo-portlet/WEB-INF/conf/portlet/webui/configuration.xml b/target/jcr-exo-portlet/WEB-INF/conf/portlet/webui/configuration.xml
new file mode 100644
index 0000000..c1d1569
--- /dev/null
+++ b/target/jcr-exo-portlet/WEB-INF/conf/portlet/webui/configuration.xml
@@ -0,0 +1,27 @@
+<!--
+
+    Copyright (C) 2003-2011 eXo Platform SAS.
+
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Affero General Public License
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see<http://www.gnu.org/licenses/>.
+
+-->
+<webui-configuration>
+  <application>
+    <ui-component-root>exoplatform.bookstore.portlet.UIBookManagementPortlet</ui-component-root>
+    <state-manager>org.exoplatform.webui.application.portlet.ParentAppStateManager</state-manager>
+    <application-lifecycle-listeners>       
+      <listener>org.exoplatform.portal.application.PortletStatisticLifecycle</listener>
+    </application-lifecycle-listeners>
+  </application>
+</webui-configuration>
\ No newline at end of file
diff --git a/target/jcr-exo-portlet/WEB-INF/portlet.xml b/target/jcr-exo-portlet/WEB-INF/portlet.xml
new file mode 100644
index 0000000..0110c5e
--- /dev/null
+++ b/target/jcr-exo-portlet/WEB-INF/portlet.xml
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2003-2011 eXo Platform SAS.
+
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Affero General Public License
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see<http://www.gnu.org/licenses/>.
+
+-->
+<portlet-app
+  xmlns="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd" version="1.0">
+  
+	<portlet>
+	    <description xml:lang="EN">Bookstore Portlet</description>
+	    <portlet-name>BookstorePortlet</portlet-name>
+	    <display-name xml:lang="EN">Bookstore Portlet</display-name>
+	    <portlet-class>org.exoplatform.webui.application.portlet.PortletApplicationController</portlet-class>
+	    <init-param>
+	      <name>webui.configuration</name>
+	      <value>/WEB-INF/conf/portlet/webui/configuration.xml</value>
+	    </init-param>
+	    <expiration-cache>0</expiration-cache>
+	    <supports>
+	      <mime-type>text/html</mime-type>
+	    </supports>
+	    <supported-locale>en</supported-locale>
+	    <portlet-info>
+	      <title>Bookstore Portlet</title>
+	      <short-title>Bookstore Portlet</short-title>
+	      <keywords>bookstore</keywords>
+	    </portlet-info>
+	    <portlet-preferences>
+	      <preference>
+	        <name>template</name>
+	        <value>app:/groovy/portlet/UIBookManagemenPortlet.gtmpl</value>
+	        <read-only>false</read-only>
+	      </preference>
+	    </portlet-preferences>
+  </portlet>    
+    
+</portlet-app>
\ No newline at end of file
diff --git a/target/jcr-exo-portlet/WEB-INF/web.xml b/target/jcr-exo-portlet/WEB-INF/web.xml
new file mode 100644
index 0000000..9455772
--- /dev/null
+++ b/target/jcr-exo-portlet/WEB-INF/web.xml
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+
+    Copyright (C) 2003-2011 eXo Platform SAS.
+
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Affero General Public License
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see<http://www.gnu.org/licenses/>.
+
+-->
+<!DOCTYPE web-app PUBLIC
+ "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN"
+ "http://java.sun.com/dtd/web-app_2_3.dtd" >
+
+<web-app>
+    
+  <display-name>BookstorePortlet</display-name>
+  
+  <description>
+    Bookstore
+  </description>
+  
+  <filter>
+    <filter-name>ResourceRequestFilter</filter-name>
+    <filter-class>org.exoplatform.portal.application.ResourceRequestFilter</filter-class>
+  </filter>
+
+  <filter-mapping>
+    <filter-name>ResourceRequestFilter</filter-name>
+    <url-pattern>/*</url-pattern>
+  </filter-mapping>
+
+  <servlet>
+    <servlet-name>GateInServlet</servlet-name>
+    <servlet-class>org.gatein.wci.api.GateInServlet</servlet-class>
+    <load-on-startup>0</load-on-startup>
+  </servlet>
+  
+  <servlet-mapping>
+    <servlet-name>GateInServlet</servlet-name>
+    <url-pattern>/gateinservlet</url-pattern>
+  </servlet-mapping>
+  
+</web-app>
diff --git a/target/jcr-exo-portlet/groovy/portlet/UIBookManagementPortlet.gtmpl b/target/jcr-exo-portlet/groovy/portlet/UIBookManagementPortlet.gtmpl
new file mode 100644
index 0000000..1d9dd63
--- /dev/null
+++ b/target/jcr-exo-portlet/groovy/portlet/UIBookManagementPortlet.gtmpl
@@ -0,0 +1,3 @@
+<div id="<% uicomponent.getId(); %>">
+	<% uicomponent.renderChildren(); %>
+</div>
\ No newline at end of file
diff --git a/target/jcr-exo-portlet/groovy/webui/UIBookList.gtmpl b/target/jcr-exo-portlet/groovy/webui/UIBookList.gtmpl
new file mode 100644
index 0000000..c79a1a5
--- /dev/null
+++ b/target/jcr-exo-portlet/groovy/webui/UIBookList.gtmpl
@@ -0,0 +1,59 @@
+<%
+	import org.exoplatform.webui.core.UIPopupWindow;
+	import org.exoplatform.portal.webui.util.Util;
+	import org.exoplatform.social.core.service.LinkProvider;
+%>
+
+<%
+	def pcontext = Util.getPortalRequestContext();
+	def jsManager = pcontext.getJavascriptManager();
+	jsManager.importJavascript("eXo.social.Util", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+	jsManager.importJavascript("eXo.social.PortalHttpRequest", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+	jsManager.importJavascript("eXo.social.webui.UIActivity", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+%>
+
+<div id="$uicomponent.id">
+
+	<%
+		def bookList = uicomponent.getBooks();
+		if (bookList.isEmpty()) {
+	%>
+		<div>No resource</div>
+	<%
+		} else {
+	%>
+			<table border="1">
+					<tr>
+						<th>Book Name</th>
+						<th>Book Category</th>
+						<th>Action</th>
+					</tr>
+				<%
+					for (book in bookList) {
+						def bookId = book.getBookId();
+						def name = book.getName();
+						def category = book.getCategory().toString();
+						def content = book.getContent();
+						def deleteBookAction = uicomponent.event("Delete", bookId) + "; return false"
+				%>
+						<tr>
+							<td>
+								$name
+							</td>
+							<td align="center">
+								$category
+							</td>
+							<td align="center">
+								<input type="button" name="btnEdit" value="Edit"/>
+								<input type="button" name="btnDelete" value="Delete" onclick="$deleteBookAction" class="Action"/>
+							</td>
+						</tr>
+				<%	
+					}
+				%>
+			</table>
+	<%
+		}
+	%>
+
+</div>
\ No newline at end of file
diff --git a/target/jcr-exo-portlet/groovy/webui/UIBookManagement.gtmpl b/target/jcr-exo-portlet/groovy/webui/UIBookManagement.gtmpl
new file mode 100644
index 0000000..e623816
--- /dev/null
+++ b/target/jcr-exo-portlet/groovy/webui/UIBookManagement.gtmpl
@@ -0,0 +1,3 @@
+<div id="$uicomponent.id">
+<% uicomponent.renderChildren(); %>
+</div>
\ No newline at end of file
diff --git a/target/jcr-exo-portlet/index.jsp b/target/jcr-exo-portlet/index.jsp
new file mode 100644
index 0000000..c38169b
--- /dev/null
+++ b/target/jcr-exo-portlet/index.jsp
@@ -0,0 +1,5 @@
+<html>
+<body>
+<h2>Hello World!</h2>
+</body>
+</html>
diff --git a/target/maven-archiver/pom.properties b/target/maven-archiver/pom.properties
new file mode 100644
index 0000000..48edded
--- /dev/null
+++ b/target/maven-archiver/pom.properties
@@ -0,0 +1,5 @@
+#Generated by Maven
+#Tue Aug 07 17:00:13 ICT 2012
+version=1.0-SNAPSHOT
+groupId=exoplatform
+artifactId=jcr-exo-portlet
